### ISSUE TITLE
Can't find bin/rackup with Rack2 binstub in place

### ISSUE THREAD

**Issue by mperham** [rack/rackup#1]
I'm running Ruby 3.1.2 and have installed a Rack 2.2 gem previously.

1. Clone the [Sidekiq repo](https://github.com/mperham/sidekiq).
2. `git co 7-0 && bundle` (this will install rack 2)
4. Adjust `sidekiq.gemspec` to use Rack `3.0.0.beta1`.
5. `cd bare && bundle up` (this will install rackup)
6. `bundle exec rackup simple.ru`

You should see a backtrace like below. I'm unclear how to fix it.

```
% bundle exec rackup simple.ru
bundler: failed to load command: rackup (/Users/mperham/.gem/ruby/3.1.2/bin/rackup)
/Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/rubygems_integration.rb:319:in `block in replace_bin_path': can't find executable rackup for gem rack (Gem::Exception)
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/rubygems_integration.rb:347:in `block in replace_bin_path'
	from /Users/mperham/.gem/ruby/3.1.2/bin/rackup:25:in `<top (required)>'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/cli/exec.rb:58:in `load'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/cli/exec.rb:58:in `kernel_load'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/cli/exec.rb:23:in `run'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/cli.rb:484:in `exec'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/cli.rb:31:in `dispatch'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/cli.rb:25:in `start'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/libexec/bundle:48:in `block in <top (required)>'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/3.1.0/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	from /Users/mperham/.rubies/ruby-3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/libexec/bundle:36:in `<top (required)>'
	from /Users/mperham/.gem/ruby/3.1.2/bin/bundle:25:in `load'
	from /Users/mperham/.gem/ruby/3.1.2/bin/bundle:25:in `<main>'
```

**Comment by ioquatix** [comment]
Did you add the rackup gem to your gemspec too?

**Comment by mperham** [comment]
No, to the Gemfile.

**Comment by mperham** [comment]
```ruby
% more Gemfile
# frozen_string_literal: true

source "https://rubygems.org"

gem "sidekiq", path: ".."
gem "rackup"
```

**Comment by ioquatix** [comment]
Interesting, does this seem like a bug in bundler/rubygems? I'll try it out too, we definitely want to streamline this process.

**Comment by ioquatix** [comment]
This should be fixed in rackup v0.2.1 - can you please test? Thanks. I missed the gemspec executables... my bad.

**Comment by mperham** [comment]
Yep, seems to work now.
